src/assets/scripts/client/game/ScoreLogic.js

```diff
@@ -0,0 +1,274 @@
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with major tailwind`, isWarning);
+        } else if (wind.head <= -5) {
+            GameController.events_recordNew(GAME_EVENTS.HIGH_TAILWIND_OPERATION);
+            UiController.ui_log(`${aircraftModel.callsign} ${action} with tailwind`, isWarning);
+        }
+    }
+
+    /**
+     * @for ScoreLogic
+     * @method _scoreRunwaySeparation
+     * @param aircraftModel {AircraftModel}
+     * @param runwayModel {RunwayModel}
+     * @param action {string}
+     */
+    _scoreRunwaySeparation(aircraftModel, runwayModel, action) {
+        const previousAircraft = this._aircraftController.findAircraftByCallsign(runwayModel.lastDepartedAircraftId);
+
+        if (previousAircraft) {
+            const actualDistance = nm_ft(aircraftModel.distanceToAircraft(previousAircraft));
+            const requiredDistance = aircraftModel.model.calculateRunwaySeparationDistanceInFeet(previousAircraft.model);
+
+            if (actualDistance < requiredDistance || previousAircraft.isOnGround()) {
+                const isWarning = true;
+
+                GameController.events_recordNew(GAME_EVENTS.NO_TAKEOFF_SEPARATION);
+                UiController.ui_log(`${aircraftModel.callsign} ${action} while another aircraft was using the same runway`, isWarning);
+            }
+        }
+    }
+
+    /**
+     * Display a waring and record an illegal glideslope intercept event
+     *
+     * @for ScoreLogic
+     * @method _penalizeLocalizerInterceptAltitude
+     * @param aircraftModel {AircraftModel}
+     */
+    _penalizeLocalizerInterceptAltitude(aircraftModel) {
+        if (aircraftModel.isAboveGlidepath()) {
+            const isWarning = true;
+
+            UiController.ui_log(`${aircraftModel.callsign} intercepted localizer above glideslope`, isWarning);
+            GameController.events_recordNew(GAME_EVENTS.LOCALIZER_INTERCEPT_ABOVE_GLIDESLOPE);
+        }
+    }
+
+    /**
+     * Display a waring and record an illegal approach event
+     *
+     * @for ScoreLogic
+     * @method _penalizeLocalizerInterceptAngle
+     * @param aircraftModel {AircraftModel}
+     */
+    _penalizeLocalizerInterceptAngle(aircraftModel) {
+        const isWarning = true;
+
+        UiController.ui_log(`${aircraftModel.callsign} approach course intercept angle was greater than 30 degrees`, isWarning);
+        GameController.events_recordNew(GAME_EVENTS.ILLEGAL_APPROACH_CLEARANCE);
+    }
+
+    /**
+     * @for ScoreLogic
+     * @method _radioCall
+     * @param msg {string}
+     * @param sectorType {string}
+     * @param isWarning {string}
+     */
+    _radioCall(aircraftModel, msg, sectorType, isWarning = false) {
+        const writtenCallsign = aircraftModel.callsign;
+        const spokenCallsign = aircraftModel.getRadioCallsign();
+
+        const logMessage = (callsign) => `${AirportController.airport_get().radio[sectorType]}, ${callsign} ${msg}`;
```

Comment:
🤔 

can we just make this a private method in this class or on the `AirportModel`?

Reply:
removed. the code is now using `AircraftModel.radioCall()`